bee-protocol/src/worker/tangle.rs

```diff
@@ -1,49 +1,51 @@
-use std::{
-    any::TypeId,
-    convert::Infallible,
-};
-use bee_common_ext::{
-    worker::Worker,
-    node::Node,
-};
+use crate::MilestoneIndex;
+
+use async_trait::async_trait;
 use bee_common::shutdown_stream::ShutdownStream;
+use bee_common_ext::{node::Node, worker::Worker};
+use bee_snapshot::metadata::SnapshotMetadata;
 use log::info;
-use async_trait::async_trait;
+use std::{any::TypeId, convert::Infallible};
 
-use crate::{
-    worker::storage::StorageWorker,
-    tangle::MsTangle,
-};
+use crate::{tangle::MsTangle, worker::storage::StorageWorker};
 
 pub struct TangleWorker;
 
 #[async_trait]
 impl<N: Node> Worker<N> for TangleWorker {
-    type Config = ();
+    type Config = SnapshotMetadata;
     type Error = Infallible;
 
     fn dependencies() -> &'static [TypeId] {
         Box::leak(Box::from(vec![TypeId::of::<StorageWorker>()]))
     }
 
-    async fn start(
-        node: &mut N,
-        config: Self::Config,
-    ) -> Result<Self, Self::Error> {
-
+    async fn start(node: &mut N, config: Self::Config) -> Result<Self, Self::Error> {
         let storage = node.storage().clone();
         let tangle = MsTangle::<N::Backend>::new(storage);
 
         node.register_resource(tangle);
 
-        let tangle = node.resource::<MsTangle::<N::Backend>>().clone();
+        let tangle = node.resource::<MsTangle<N::Backend>>().clone();
         node.spawn::<Self, _, _>(|shutdown| async move {
-            use tokio::time::interval;
-            use std::time::Duration;
             use futures::StreamExt;
+            use std::time::Duration;
+            use tokio::time::interval;
 
             let mut receiver = ShutdownStream::new(shutdown, interval(Duration::from_secs(1)));
 
+            tangle.update_latest_solid_milestone_index(config.index().into());
```

Comment:
Should this be done outside the task? If it's done inside the task, dependent workers may end up with an incomplete view of the tangle.

Reply:
Done